app/models/billable_metric.rb

```diff
@@ -24,6 +24,8 @@ class BillableMetric < ApplicationRecord
 
   enum aggregation_type: AGGREGATION_TYPES
 
+  validate :validate_recurring, if: -> { count_agg? || max_agg? || recurring_count_agg? }
```

Comment:
It fill a bit weird to have the preconditions splitter between this lambda and the validation method. What do you think of moving all conditions into the method?

Reply:
👍